### PR TITLE
Fixed the service ca deployment issue and refactor the test setup structure

### PR DESCRIPTION
**Pull request by yanmxa**
Signed-off-by: Meng <yanmxa@163.com>

### CODE REVIEW

**Comment by yanmxa** [comment]
https://github.com/stolostron/backlog/issues/23507

File: test/setup/common.sh [hunk 1]
```diff
@@ -56,12 +55,10 @@ function checkKind() {
 
 function initKinDCluster() {
   clusterName=$1
-  while [[ $(kind get clusters | grep "^${clusterName}$") != "${clusterName}" ]]; do
-    kind create cluster --name "$clusterName" --image kindest/node:v1.23.4 --wait 40s
-  done
-  # enable the applications.app.k8s.io
-  kubectl config use-context "kind-${clusterName}"
-  kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/application/master/deploy/kube-app-manager-aio.yaml
+  image="kindest/node:v1.23.6@sha256:b1fa224cc6c7ff32455e0b1fd9cbfd3d3bc87ecaa8fcb06961ed1afb3db0f9ae" # or kindest/node:v1.23.4
```
Comment (yanmxa) [inline test/setup/common.sh:58]: This version of the image uses fewer memory resource.

**Comment by clyang82** [comment]
💯 
/lgtm

**Pull request merged**